tests/drivers/mspi/error_cases/src/main.c

```diff
@@ -0,0 +1,686 @@
+}
+
+ZTEST(mspi_error_cases, test_01a_MSPI_OP_MODE_CONTROLLER)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.op_mode = MSPI_OP_MODE_CONTROLLER;
+
+	check_mspi_config(&hw_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_01b_MSPI_OP_MODE_PERIPHERAL)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.op_mode = MSPI_OP_MODE_PERIPHERAL;
+
+	check_mspi_config(&hw_cfg, -ENOTSUP);
+}
+
+ZTEST(mspi_error_cases, test_02a_MSPI_HALF_DUPLEX)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.duplex = MSPI_HALF_DUPLEX;
+
+	check_mspi_config(&hw_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_02b_MSPI_FULL_DUPLEX)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.duplex = MSPI_FULL_DUPLEX;
+
+	check_mspi_config(&hw_cfg, -ENOTSUP);
+}
+
+ZTEST(mspi_error_cases, test_03a_dqs_support_true)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.dqs_support = true;
+
+	check_mspi_config(&hw_cfg, -ENOTSUP);
+}
+
+ZTEST(mspi_error_cases, test_03b_dqs_support_false)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.dqs_support = false;
+
+	check_mspi_config(&hw_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_04a_sw_multi_periph_true)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.sw_multi_periph = true;
+
+	check_mspi_config(&hw_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_05a_num_ce_gpios_max)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.num_ce_gpios = UINT32_MAX;
+
+	check_mspi_config(&hw_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_06a_num_periph_max)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.num_periph = UINT32_MAX;
+
+	check_mspi_config(&hw_cfg, -EINVAL);
+}
+
+ZTEST(mspi_error_cases, test_07a_max_freq_max)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.max_freq = UINT32_MAX;
+
+	check_mspi_config(&hw_cfg, -ENOTSUP);
+}
+
+ZTEST(mspi_error_cases, test_08a_re_init_true)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.re_init = true;
+
+	check_mspi_config(&hw_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_08b_re_init_false)
+{
+	struct mspi_cfg hw_cfg = default_hw_cfg;
+
+	hw_cfg.re_init = false;
+
+	check_mspi_config(&hw_cfg, 0);
+}
+
+void check_mspi_dev_config(const enum mspi_dev_cfg_mask param_mask,
+		const struct mspi_dev_cfg *cfg,
+		int expected)
+{
+	const struct mspi_dt_spec spec = {
+		.bus    = mspi_bus,
+		.config = default_hw_cfg,
+	};
+
+	ret = mspi_config(&spec);
+	zassert_equal(ret, 0, "Got unexpected %d instead of 0", ret);
+
+	ret = mspi_dev_config(mspi_bus, &dev_id[0], param_mask, cfg);
+	zassert_equal(ret, expected, "Got unexpected %d instead of %d", ret, expected);
+}
+
+ZTEST(mspi_error_cases, test_09a_MSPI_DEVICE_CONFIG_NONE)
+{
+	check_mspi_dev_config(MSPI_DEVICE_CONFIG_NONE, &device_cfg[0], 0);
+}
+
+ZTEST(mspi_error_cases, test_10a_MSPI_DEVICE_CONFIG_CE_NUM_MAX)
+{
+	struct mspi_dev_cfg dev_cfg = device_cfg[0];
+
+	dev_cfg.ce_num = UINT8_MAX;
+
+	check_mspi_dev_config(MSPI_DEVICE_CONFIG_CE_NUM, &dev_cfg, 0);
```

Comment:
```suggestion
	dev_cfg.ce_num = UINT8_MAX;

	check_mspi_dev_config(MSPI_DEVICE_CONFIG_CE_NUM, &dev_cfg, -EINVAL);
```
For now only five CE pins are supported.